parser_advanced_test.go

```diff
@@ -0,0 +1,386 @@
+package participle_test
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"text/scanner"
+
+	require "github.com/alecthomas/assert/v2"
+	"github.com/alecthomas/participle/v2"
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+type operatorPrec struct{ Left, Right int }
+
+var operatorPrecs = map[string]operatorPrec{
+	"+": {1, 1},
+	"-": {1, 1},
+	"*": {3, 2},
+	"/": {5, 4},
+	"%": {7, 6},
+}
+
+type (
+	Expr interface{ expr() }
+
+	ExprIdent  struct{ Name string }
+	ExprString struct{ Value string }
```

Comment:
Not `ExprString string`?

Reply:
Ah tbh I wasn't really thinking about it, I was just declaring structs. I can make it a string!

Reply:
I was lazy and didn't change these, but I did move it to the `_examples/expr4` directory as an example of custom expression parsing.